docs/spec/spec.md

```diff
@@ -0,0 +1,345 @@
+# Specification: Ballerina WebSocket Library
+
+_Owners_: @shafreenAnfar @bhashinee
+_Reviewers_: @shafreenAnfar    
+_Created_: 2021/12/09  
+_Updated_: 2021/12/09  
+_Issue_: [#2165](https://github.com/ballerina-platform/ballerina-standard-library/issues/2165)
+
+# Introduction
+
+This is the specification for WebSocket standard library which is used to implement WebSocket compliant 'listener' and 'client' using [Ballerina programming language](https://ballerina.io/), which is an open-source programming language for the
+cloud that makes it easier to use, combine, and create network services. 
+
+# Contents
+1. [Overview](#1-overview)
+2. [Listener](#2-listener)
+3. [Service Types](#3-service-types)
+   * 3.1. [Upgrade Service](#31-upgrade-service)
+   * 3.2. [WebSocket Service](#32-websocket-service)
+4. [Client](#4-client)
+5. [Securing the WebSocket Connections](#5-securing-the-websocket-connections)
+   * 5.1. [SSL/TLS](#51-ssl-tls)
+   * 5.2. [Authentication and Authorization](#52-authentication-and-authorization)
+6. [Samples](#6-samples)
+
+## 1. [Overview](#1-overview)
+
+WebSocket is a protocol that allows a long held full-duplex connection between a server and client. This specification elaborates on how Ballerina language provides a tested WebSocket client and server implementation that is compliant with the [RFC 6455](https://www.rfc-editor.org/rfc/rfc6455.html).
+
+## 2. [Listener](#2-listener)
+
+The WebSocket listener can be constructed with a port or an `http:Listener`. When initiating the listener it opens up the port and attaches the upgrade service which is quite similar to an HTTP service at the given service path. 
+
```

Comment:
The WebSocket listener can be constructed with a port or an `http:Listener`. When initiating the listener it opens up the port and attaches the upgrade service at the given service path. It is also worth noting that upgrade service is quite similar to an HTTP service.